keps/sig-network/20190603-EndpointSlice-API.md

```diff
@@ -0,0 +1,273 @@
+---
+title: EndpointSlice API 
+authors:
+  - "@freehan"
+owning-sig: sig-network
+reviewers:
+  - "@bowei"
+  - "@thockin"
+  - "@wojtek-t"
+  - "@johnbelamaric"
+approvers:
+  - "@bowei"
+  - "@thockin"
+creation-date: 2019-06-01
+last-updated: 2019-06-01
+status: implementable
+see-also:
+  - "https://docs.google.com/document/d/1sLJfolOeEVzK5oOviRmtHOHmke8qtteljQPaDUEukxY/edit#"
+---
+# EndpointSlice API 
+
+## Summary 
+
+This KEP was converted from the [original proposal doc][original-doc]. The current  [Core/V1 Endpoints API][v1-endpoints-api] comes with severe performance/scalability drawbacks affecting multiple components in the control-plane (apiserver, etcd, endpoints-controller, kube-proxy). 
+This doc proposes a new EndpointSlice API aiming to replace Core/V1 Endpoints API for most internal consumers, including kube-proxy.
+The new EndpointSlice API aims to address existing problems as well as leaving room for future extension.
+
+
+## Motivation
```

Comment:
Can you explain the use case for having multiple megabytes of endpoints for a single service in the first place?